packages/cli/src/zosfiles/compare/lf-ds/LocalfileDataset.handler.ts

```diff
@@ -39,9 +39,11 @@ export default class LocalfileDatasetHandler extends ZosFilesBaseHandler {
             localFile = path.resolve(commandParameters.arguments.localFilePath);
         }
 
+        let localFileHandle: number;
         // check if the path given is of a file or not
         try {
-            if(!fs.lstatSync(localFile).isFile()){
+            localFileHandle = fs.openSync(localFile, 'r');
+            if(!fs.fstatSync(localFileHandle).isFile()){
```

Comment:
I think this is currently being updated in the other CodeQL QRs, but we probably want to use a `finally` block here - (as you suggested earlier) to call `closeSync` in scenarios where `fstatSync` fails, but the file is still opened